crates/swc_ecma_transforms_compat/src/es2020/opt_chaining.rs

```diff
@@ -222,58 +228,58 @@ impl OptChaining {
             }
         }
 
-        Expr::Unary(e)
+        Expr::Unary(e.take())
     }
 
-    /// Only called from [Fold].
-    fn handle_call(&mut self, e: CallExpr) -> Result<CondExpr, Expr> {
-        match e.callee {
+    /// Only called from [VisitMut].
+    fn handle_call(&mut self, e: &mut CallExpr) -> Result<CondExpr, Expr> {
+        match &mut e.callee {
             ExprOrSuper::Expr(callee) if callee.is_opt_chain() => {
-                let callee = callee.opt_chain().unwrap();
-                let expr = self.unwrap(callee);
+                let mut callee = (&mut *callee).take().opt_chain().unwrap();
+                let expr = self.unwrap(&mut callee);
 
                 return Ok(CondExpr {
                     span: DUMMY_SP,
                     alt: Box::new(Expr::Call(CallExpr {
                         callee: ExprOrSuper::Expr(expr.alt),
-                        ..e
+                        ..e.take()
                     })),
-                    ..expr
+                    test: expr.test,
```

Comment:
Seems like unnecessary change?

Reply:
Yes, thanks for pointing out. lot of these are leftover while I made changes on top of each, changing one and forgot to bring it back.